### DISCUSSION TITLE
Properly detect tags

### DISCUSSION THREAD

**Discussion by tomzx** [damionx7/Zettel-Notes-Documentation#87] [Feature Requests]
Text which contains tags but are not prefixed by a space are not properly detected. 
Examples:
* (#atag)
* [#atag]
* {#atag}

**Comment by damionx7** [comment]
Hey! Zettel Notes tags need to have a space in front to distinguish from surrounding text. Also, only the second tag works in Obsidian [#atag]. Can you please tell me which application you are using?

**Reply by tomzx** [reply]
What is the current regex used to do tag matching? If you only mind supporting tags with ASCII characters we could use `#\w+` to match hash tags, alternatively [`#\p{L}+`](https://regex101.com/r/TLc3oK/1) seems to work with Unicode.

This would definitely start matching hash tags in URLs as there would be no requirement for a tag to start with a space (or start the document), making it a breaking change.

**Reply by damionx7** [reply]
I use the following regex `(?<!\S)([#][\w\-\/]+)` for matching tags. Basically this prevents matching tags in words or as you mentioned URLs.
